docs/docs/how-to/images-and-media/using-gatsby-plugin-image.md

```diff
@@ -0,0 +1,161 @@
+
+### Static images
+
+If you are using an image that will be the same each time the component is used, such as a logo or hero image, you can use the `StaticImage` component. The image can be either a local file in your project, or an image hosted on a remote server. Any remote images are downloaded and resized at build time.
+
+1. **Add the image to your project**
+
+   If you are using a local image, copy it into the project. A folder such as `src/images` is a good choice.
+
+2. **Add the `StaticImage` component to your template**
+
+   Import the component, then set the `src` prop to point to the image you added earlier, or to the URL of the image if hosted elsewhere. The path is relative to the source file itself. If your component file was `src/components/dino.js`, then you would load the image like this:
+
+   ```jsx
+   import { StaticImage } from "gatsby-plugin-image"
+
+   export function Dino() {
+     return (
+       <section>
+         <StaticImage src="../images/dino.png" alt="A dinosaur" />
+         <StaticImage src="https://placekitten.com/800/600" alt="A kitten" />
+       </section>
+     )
+   }
+   ```
+
+   When you build your site, this will load the image from your filesystem or from the remote URL, and generate all the sizes and formats that you need to support a responsive image. Because the image is loaded at build time, you cannot pass the filename in as a prop, or otherwise generate it outside of the component. It should either be a static string, or a local variable in the component's scope.
+
+   > **Important:** remote images are downloaded and resized at build time. If the image is changed on the other server, it will not be updated on your site until you rebuild
+
+3. **Configure the image**:
+
+   You configure the image by passing props to the `<StaticImage />` component. You can change the size and layout, as well as settings such as the type of placeholder used when lazy loading. There are also advanced image processing options available. You can find the full list of options in the API docs.
+
+   ```jsx
+   import { StaticImage } from "gatsby-plugin-image"
+
+   export function Dino() {
+     return (
+       <StaticImage
+         src="../images/dino.png"
+         alt="A dinosaur"
+         placeholder="blurred"
+         layout="fixed"
+         width={200}
+         height={200}
+       />
+     )
+   }
+   ```
+
+### Dynamic images
+
+If you need to have dynamic images, such as if they are coming from a CMS, they are loaded via GraphQL and displayed using the `GatsbyImage` component.
+
+1. **Add the image to your page query**:
+
+   Any GraphQL File object that includes an image will have a `childImageSharp` field that you can use to query the image data. The exact data structure will vary according to your data source, but the syntax is like this:
+
+   ```graphql
```

Comment:
For the three dynamic image examples, could you add a title to the code blocks of what file these might live in?

(I don't have super strong feelings about it one way or the other, but it might be helpful to newbies.)

Reply:
Will do